fearless/Modules/Staking/StakingMain/StakingMainInterator+Subscription.swift

```diff
@@ -314,204 +130,171 @@ extension StakingMainInteractor {
-                chain: chain,
-                runtimeService: runtimeService
-            )
-        }
+    func handleMaxNominatorsCount(result: Result<UInt32?, Error>, chainId _: ChainModel.Id) {
+        presenter.didReceiveMaxNominatorsCount(result: result)
     }
+}
 
-    private func fetchAnalyticsRewards(stash: AccountAddress) {
-        guard let analyticsURL = currentConnection?.type.chain.analyticsURL else { return }
-        let period = analyticsPeriod
-
-        let now = Date().timeIntervalSince1970
-        let sevenDaysAgo = Date().addingTimeInterval(-(.secondsInDay * 7)).timeIntervalSince1970
-        let subqueryRewardsSource = SubqueryRewardsSource(
-            address: stash,
-            url: analyticsURL,
-            startTimestamp: Int64(sevenDaysAgo),
-            endTimestamp: Int64(now)
-        )
-        let fetchOperation = subqueryRewardsSource.fetchOperation()
-
-        fetchOperation.targetOperation.completionBlock = { [weak self] in
-            DispatchQueue.main.async {
-                do {
-                    let response = try fetchOperation.targetOperation.extractNoCancellableResultData()
-                    self?.presenter?.didReceieve(subqueryRewards: .success(response), period: period)
-                } catch {
-                    self?.presenter?.didReceieve(subqueryRewards: .failure(error), period: period)
-                }
+extension StakingMainInteractor: PriceLocalStorageSubscriber, PriceLocalSubscriptionHandler {
+    func handlePrice(result: Result<PriceData?, Error>, priceId: AssetModel.PriceId) {
+        if let chainAsset = stakingSettings.value, chainAsset.asset.priceId == priceId {
+            switch result {
+            case let .success(priceData):
+                presenter.didReceive(price: priceData)
+            case let .failure(error):
+                presenter.didReceive(priceError: error)
             }
         }
-        operationManager.enqueue(operations: fetchOperation.allOperations, in: .transient)
     }
 }
 
-extension StakingMainInteractor: SingleValueProviderSubscriber, SingleValueSubscriptionHandler,
-    SubstrateProviderSubscriber, SubstrateProviderSubscriptionHandler,
-    AnyProviderAutoCleaning {
-    var singleValueProviderFactory: SingleValueProviderFactoryProtocol {
-        providerFactory
-    }
-
-    func handleMinNominatorBond(result: Result<BigUInt?, Error>, chain _: Chain) {
-        presenter.didReceiveMinNominatorBond(result: result)
+extension StakingMainInteractor: WalletLocalStorageSubscriber, WalletLocalSubscriptionHandler {
+    func handleAccountInfo(
+        result: Result<AccountInfo?, Error>,
+        accountId _: AccountId,
+        chainId _: ChainModel.Id
+    ) {
+        switch result {
+        case let .success(accountInfo):
+            presenter.didReceive(accountInfo: accountInfo)
+        case let .failure(error):
+            presenter.didReceive(balanceError: error)
```

Comment:
The label `balanceError:` is misleading. I suggest to rename it according to its counterpart to `accountInfoError`. (I know, the better way would be to refactor everything to receive `Result` etc.)